asapdiscovery-spectrum/asapdiscovery/spectrum/score_complex.py

```diff
@@ -0,0 +1,363 @@
+
+        # Default pre-min score if not found
+        pre_min_score = 0
+
+        if prot_id is not None:
+            try:
+                # 1 ligand, multiple targets
+                pre_min_score = df_dock.loc[
+                    df_dock["prot-ID"] == prot_id, "docking-score-POSIT"
+                ].iloc[0]
+            except (IndexError, KeyError):
+                logging.warning("No pre-docking score found for protein %s", prot_id)
+            tag = prot_id
+        else:
+            tag = ""
+
+        if ligand is not None:
+            try:
+                # 1 target, multiple ligands
+                pre_min_score = df_dock.loc[
+                    df_dock["lig-ID"] == ligand, "docking-score-POSIT"
+                ].iloc[0]
+            except (IndexError, KeyError):
+                logging.warning("No pre-docking score found for ligand %s", ligand)
+
+            if prot_id is not None:
+                # multiple targets, multiple ligands
+                try:
+                    pre_min_score = df_dock.loc[
+                        (df_dock["lig-ID"] == ligand) & (df_dock["prot-ID"] == prot_id),
+                        "docking-score-POSIT",
+                    ].iloc[0]
+                except (IndexError, KeyError):
+                    logging.warning(
+                        "No pre-docking score found for ligand %s and protein %s",
+                        ligand,
+                        prot_id,
+                    )
+                tag += "_"
+
+            tag += f"{ligand}"
+
+        logging.info(
+            "Scoring protein %s and ligand %s from file %s", prot_id, ligand, file_min
+        )
+        # Reference structure
+        if path_ref.is_dir():
+            files_in_dir = list(path_ref.glob(f"*{ligand}*.pdb"))
+            if len(files_in_dir) > 0:
+                file_ref = files_in_dir[0]  # return first find
+                logging.info("The ref %s was found for %s", file_ref, tag)
+            else:
+                logging.error("A reference was not found for %s", tag)
+                continue
+        else:
+            file_ref = path_ref
+
+        # Run minimization if requested
+        min_out = file_min
+        if inputs.minimize:
+            min_folder = docking_dir / "minimized"
+            min_folder.mkdir(parents=True, exist_ok=True)
+            new_docking_dir = min_folder
+            md_openmm_platform = inputs.md_openmm_platform
+            try:
+                min_out = f"{min_folder}/{tag}_min.pdb"
+                logging.info("Running MD minimization of %s", tag)
+                minimize_structure(
+                    file_min,
+                    min_out,
+                    min_folder,
+                    md_openmm_platform,
+                    inputs.target,
+                    comp_name,
+                )
+                chain_dock = "1"  # Standard in OpenMM output file
+            except Exception as error:
```

Comment:
We should avoid catching generic exceptions. Any specific exceptions here that we want to catch?